monasca_events_api/tests/test_transforms.py

```diff
@@ -0,0 +1,347 @@
+        helper_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        transform_id = "0ab1ac0a"
+        generateUUID.return_value = "067e6162-3b6f-4ae2-a171-2470b63dff00"
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertRaises(falcon.HTTPInternalServerError)
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._validate_transform')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    def test_on_post_fail_validate_transform(
+            self,
+            readhttp,
+            helpers_validate,
+            validjson,
+            _validate_transform):
+        """Post Method fail due to validate transform"""
+        helpers_validate.return_value = True
+        validjson.return_value = True
+        _validate_transform.side_effect = falcon.HTTPBadRequest(
+            'Bad request',
+            'Error')
+        readhttp.return_value = self._generate_req()
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertRaises(falcon.HTTPBadRequest)
+            self.assertEqual(e.status, '400 Bad Request')
+
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    def test_on_post_fail_validate_authorization(
+            self,
+            _validate_authorization):
+        """Post Method fail due to validate authorization"""
+        _validate_authorization.side_effect = falcon.HTTPUnauthorized(
+            'Forbidden',
+            'Tenant does not have any roles')
+
+        transObj = TransformsSubClass()
+        transObj._transforms_repo = TransformsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            transObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertRaises(falcon.HTTPUnauthorized)
+            self.assertEqual(e.status, '401 Unauthorized')
+
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._create_transform_response')
+    @mock.patch(
+        'monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.openstack.common.uuidutils.generate_uuid')
+    @mock.patch(
+        'monasca_events_api.v2.transforms.Transforms._validate_transform')
```

Comment:
Can you provide some correct data cause after all we are just testing with stubbed data.